### PR TITLE
Fix response structures in mediator coordination protocol

### PR DESCRIPTION
**Pull request by Hermann-Core**

### CODE REVIEW

**Review by Christiantyemele — APPROVED** [review]

**Comment by Christiantyemele** [comment]
@Hermann-Core core you have some conflicts though

**Review by ndefokou — APPROVED** [review]
Good

**Review by Blindspot22 — APPROVED** [review]
`mediate_deny` and `mediate_grant` are structured correctly, adhering to the protocol's expectations. So i think everything is good with regards to the ticket aim.

File: crates/plugins/mediator-coordination/src/model/stateful/coord.rs [hunk 1]
```diff
@@ -279,22 +258,21 @@ pub struct KeylistBody {
...
     pub offset: i32,
     pub remaining: i32,
 }
+
+/// Message to grant a routing DID to a mediator.
 #[derive(Debug, Serialize, Deserialize, Clone, Default)]
-#[serde(rename_all = "snake_case")]
 pub struct MediationGrantBody {
     pub routing_did: String,
 }
```
Comment (IngridPuppet) [inline crates/plugins/mediator-coordination/src/model/stateful/coord.rs:278]: This does not sound valid a description to me. The mediator, as the message sender, does not grant a routing DID to itself. The grant message is instead an agreement to mediate a requesting agent's messages.

**Review by IngridPuppet — CHANGES_REQUESTED** [review]

File: crates/plugins/mediator-coordination/src/model/stateful/coord.rs (hunk 1 above)
Comment (Hermann-Core) [reply crates/plugins/mediator-coordination/src/model/stateful/coord.rs:278]: I don’t know how this comment appeared here, maybe a dumb copy paste. I will fix that.

**Review by IngridPuppet — APPROVED** [review]

**Pull request merged**